2_munge/src/aggregate_inputs.R

```diff
@@ -125,7 +153,7 @@ aggregate_nwm <- function(ind_file, raw_ind_file, remake_file, sites_file, gd_co
       select(contains('00:00:00')) %>% # we only want ref dates that start at midnight
       dplyr::do(with(., {
         ref_dates = rep(as.Date(colnames(.), tz = 'UTC'), each = nrow(.))
-        n_ref_dates = length(unique(ref_dates))
+        n_ref_dates = length(unique(ref_dates)) # is unique() the right logic here? or does it only work because we've already selected down to unique dates?
```

Comment:
@jzwart - i was wondering about this call to `unique()` as I tried to work out similar code for the analysis predictions. could this just be `length(ref_dates)` here?

Reply:
Hmmm. I think it needs `unique()` because it's referencing the line above which replicates `ref_dates` by the number of rows as I was turning it into a long table format. So without `unique` I think there we'd end up with 3 times too many `ref_dates`, but I haven't tried the code out yet. I also wrote this before I knew the `gather` function and a lot of this can probably be replaced with that...

Reply:
Hmm...yep, OK, I see and agree that it must work fine here. The problem I was hitting for the analysis data was that there were multiple columns with the same as.Date() value, so unique() was getting us down to too few values, but that doesn't apply here because you're selecting just the midnight values anyway.

Reply:
oh yeah that makes sense for analysis data.